File: Unit2-03/Java/MrCoxallStack.java
/*
* This class defines a string stack
*
* @author  Kenny Le
* @version 1.0
* @since   2024-03-08
*/

import java.util.ArrayList;
import java.util.List;

/**
 * This is the class.
*/
public class MrCoxallStack {
    /**
     * Adds strings to an array.
     *
     * @param array contains the array of strings
    */
    private List<String> stackAsList;

    /**
     * Constructs a new instance of MrCoxallStack.
     * Initializes the stack with an empty ArrayList.
     *
    */
    public MrCoxallStack() {
        this.stackAsList = new ArrayList<>();
    }

    // getters

    /**
     * The getStackAsList() getter.
     *
     * @return -  current list
     */
    public List<String> getStackAsList() {
        return this.stackAsList;
    }

    /**
     * Pushes the string to a list.
     *
    */
    public void push(String stringPushed) {
        this.stackAsList.add(stringPushed);
    }

    /**
     * Pop the top item.
     *
     * @return the popped item from the stack
    */

    public String pop() {
        String returnValue = null;
        if (this.stackAsList.size() > 0) {
            final String topItem = this.stackAsList.get(this.stackAsList.size() - 1);
            returnValue = topItem;
            this.stackAsList.remove(topItem);
        } else {
            System.out.println("The Stack is empty!");
        }

        return returnValue;
    }

    /**
     * Returns the stack value.
     *
     * @return the values of the list formatted
    */
    public String showStack() {
        final StringBuilder stackValues = new StringBuilder();

        for (String value : this.stackAsList) {
            stackValues.append(value).append(", ");
        }

        if (stackValues.length() > 0) {
            // Remove the trailing comma
            stackValues.deleteCharAt(stackValues.length() - 2);
        }

        return stackValues.toString();
    }
}

File: Unit2-03/Typescript/MrCoxallStack.ts
/**
* This class defines a string stack
*
* By:      Kenny Le
* Version: 1.0
* Since:   2024-03-07
*/

export class MrCoxallStack {
    // Stack that contains all of the string
    private stackAsList: string[]
  
    // Creates the stack
    public constructor() {
      this.stackAsList = []
    }
  
    // Getter for stackAsList
    get stackAsList() {
      return this._stackAsList
    }
  
    // Pushes the values to the stack
    public push(stringPushed: string) {
      this.stackAsList.push(stringPushed)
    }
  
    // pop the value from the stack
    public pop() {
      let returnValue: string = ""
      if (this.stackAsList.length > 0) {
        let topItem = this.stackAsList[this.stackAsList.length - 1]
        this.stackAsList.length = this.stackAsList.length - 1
        returnValue = topItem
      } else {
        returnValue = "The stacks is empty!"
      }
  
      return returnValue
    }
  
    // Returns the stack
    public showStack() {
      let stackItems = this.stackAsList[0]
  
      for (let counter = 1; counter < this.stackAsList.length; counter++) {
        stackItems = stackItems + ", " + this.stackAsList[counter]
      }
  
      return stackItems
    }
  }
  
  